dataprofiler/data_readers/base_data.py

```diff
@@ -172,3 +172,33 @@ def length(self):
         :return: length of the dataset
         """
         return len(self)
+
+    def __getattribute__(self, name):
+        """
+        Overrides getattr for the class such that functions can be applied
+        directly to the data class if the function is not part of the data
+        class.
+        e.g. if data is BaseData where self.data = [1, 2, 3, 1]
+        ```
+        data.count(1)  # returns 2, bc data.data has the function 'count'
+        ```
+        """
+        try:
+            returned = object.__getattribute__(self, name)
+        except AttributeError as attr_error:
+            class_name = self.__class__.__name__
+            data_class_name = self.data.__class__.__name__
+            if (not f"'{class_name}' object has no attribute '{name}'"
+                    == str(attr_error)):
```

Comment:
Its seems strange to use error string for the conditional, but I'm not sure any other way to do this check

Reply:
i validate it is an attribute error with the except and then validate the string, there might be a better way.